src/dmd/parse.d

```diff
@@ -6029,19 +6029,56 @@ final class Parser(AST) : Lexer
         switch (token.value)
         {
         case TOK.leftCurly:
-            /* Scan ahead to see if it is a struct initializer or
-             * a function literal.
-             * If it contains a ';', it is a function literal.
-             * Treat { } as a struct initializer.
+            /* Scan ahead to discern between struct initializer and
+             * parameterless function literal.
+             *
+             * We'll scan the topmost curly bracket level for statement-related
+             * tokens, thereby ruling out a struct initializer.  (A struct
+             * initializer which itself containins function literals may have
+             * statements at nested curly bracket levels.)
+             *
+             * It's important that this functional literal check not be
+             * pendantic, otherwise a function having the slightest syntax
+             * error would emit confusing errors when we proceed to parse it
+             * as a struct initializer.
```

Comment:
This paragraph doesn't help the reader as that's really obvious. I suggest to drop it.

Reply:
[copied from other thread] This matter was indeed not obvious to me so seems worthy of source code comment. If the author intended a function literal (implies heuristics as employed here), then even when that literal has a syntax error we want to parse it as a function and emit understandable errors.